### PR TITLE
Implement DEP 12, string literal syntax

### PR DESCRIPTION
**Pull request by cgay**
https://opendylan.org/proposals/dep-0012-string-literals.html

### CODE REVIEW

File: documentation/library-reference/source/language-extensions/string-literals.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+`Dylan Enhancement Proposal (DEP) 12
+<https://opendylan.org/proposals/dep-0012-string-literals.html>`_ extends the
+literal syntax for strings to include strings without any escape character
+processing ("raw" strings) and multi-line strings. Briefly,
+
+#. Multi-line strings begin with three double-quote characters: ``"""``
+
+#. End-of-line sequences in multi-line strings are always parsed as a single
+   Newline (``\\n``) character, regardless of operating system.
```
Comment (housel) [inline documentation/library-reference/source/language-extensions/string-literals.rst:12]: regardless of source file line endings or the conventions of the operating system.

File: sources/dfmc/reader/lexer.dylan [hunk 2]
```diff
@@ -365,20 +368,55 @@ define method get-token
...
+    do-process-token(kind, lexer, source-location)
+  elseif (unexpected-eof?)
+    invalid-end-of-input(source-location);
+  else
+    invalid-token(source-location);
+  end if
+end method get-token;
+
+// This is separated out from get-token so as to be testable without
+// having to make a <lexer>, which inturn requires having to pull in
```
Comment (housel) [inline sources/dfmc/reader/lexer.dylan:397]: in turn

**Review by housel — APPROVED** [review]

File: documentation/library-reference/source/language-extensions/index.rst [hunk 3]
```diff
@@ -44,10 +44,9 @@ convenient "dialect" of Dylan, exported from the module ``common-dylan``:
       use common-extensions, export: all;
     end module;
 
-This section describes the common language extensions, that is,
-extensions made to the Dylan library as it is defined in DRM. These
-extensions are available to applications in the ``dylan`` library's
-``dylan`` module.
+This section describes Dylan language extensions, that is, extensions made to
+the Dylan library as it is defined in the DRM. These extensions are available
```
[the remark this answers is not in the document]
Comment (cgay) [reply documentation/library-reference/source/language-extensions/index.rst]: Done

File: sources/dfmc/reader/lexer.dylan (hunk 2 above)
Comment (cgay) [reply sources/dfmc/reader/lexer.dylan]: Done

File: documentation/library-reference/source/language-extensions/string-literals.rst (hunk 1 above)
Comment (cgay) [reply documentation/library-reference/source/language-extensions/string-literals.rst]: Done

**Comment by cgay** [comment]
Ignore this comment. Just testing comments on a PR to see if they show up in Matrix room.

**Pull request merged**